### PR TITLE
Event / University Coordinate Fetching

### PR DESCRIPTION
**Pull request by janto393**
Implemented functionality for the `getEvents` and `getUniversities` endpoints where the geolocation coordinates would be returned with the data.

Also fixed the following bugs:
- Error in `getUniversities` where the response was not being sent
- Error in `getUniversities` where the query was joining the states table on the wrong columns

The front-end types were also updated to include the new data

### CODE REVIEW

File: src/api/getEvents.ts [hunk 1]
```diff
@@ -98,10 +95,14 @@ interface Event
 	numAttendees: number,
 	capacity: number,
 	eventPictures: Array<EventPicture>,
-	eventLat: number,			 	// find this
-	eventLong: number, 		 	// find this 
-	schoolLat: number, 			// find this
-	schoolLong: number 			// find this
+	lattitude: number,			 	// find this
```
Comment (jamilgonzalez) [inline src/api/getEvents.ts]: nit: latitude

File: src/api/getEvents.ts [hunk 2]
```diff
@@ -118,17 +119,44 @@ interface EndpointReturn
...
+		if (locations[0].latitude !== undefined)
+		{
+			coordinates.lattitude = locations[0].latitude;
+		}
+
+		if (locations[0].longitude !== undefined)
+		{
+			coordinates.longitude = locations[0].longitude;
+		}
+	}
```
Comment (jamilgonzalez) [inline src/api/getEvents.ts]: Probably safe to do something like

`locations.length > 0 ? locations : undefined`

or 

`const { latitude, longitude } = locations`

`latitude && longitude ? { lng: longitude, lat: latitude} : undefined`

File: src/api/getEvents.ts [hunk 3]
```diff
@@ -316,13 +347,20 @@ export async function getEvents(request: Request, response: Response, next: Call
...
-						eventLong: await getlong(String(rawData.eventAddress) + String(rawData.eventCity) + String(rawData.stateName) + String(rawData.eventZip)),
-						schoolLat: await getlat(String(rawData.schoolAddress) + String(rawData.schoolCity) + String(rawData.schoolStateName) + String(rawData.schoolZip)),
-						schoolLong: await getlong(String(rawData.schoolAddress) + String(rawData.schoolCity) + String(rawData.schoolStateName) + String(rawData.schoolZip))
+						// set coordinates to zero (will calculate after event creation)
+						lattitude: 0,
+						longitude: 0
 					};
+
+					// get event Coordinates
+					let eventCoords: Coordinates = await getLattitudeAndLongitude(event.address, event.city, event.state.name, event.zip);
```
Comment (jamilgonzalez) [inline src/api/getEvents.ts:273]: nit: same as comment above about latitude spelling

**Comment by jamilgonzalez** [comment]
Is there any progress on this PR? Comments have been sitting for a week.

**Comment by janto393** [comment]
> 
> 
> Is there any progress on this PR? Comments have been sitting for a week.

I didn't see the comments in this PR. I'll work on it tonight and get it fixed.

File: src/api/getEvents.ts (hunk 2 above)
Comment (janto393) [reply src/api/getEvents.ts]: Implemented safer practice:
```typescript
	// parse the data of the first match
	if ((locations !== undefined) && (locations.length > 0))
	{
		const { latitude, longitude } = locations[0];

		coordinates.latitude = latitude;
		coordinates.longitude = longitude;
	}
```

**Comment by janto393** [comment]
Fixed latitude spelling across all API endpoints. Packaged latitude and longitude into `Coordinate` type.

Fixed the fields passing into the `getUniversities` and `getEvents` endpoints from `schoolID` to `universityID`. Updated front-end types and API calls to accommodate change.

Implemented safer practice of location parsing:
```typescript
// parse the data of the first match
	if ((locations !== undefined) && (locations.length > 0))
	{
		const { latitude, longitude } = locations[0];

		coordinates.latitude = latitude;
		coordinates.longitude = longitude;
	}
```

**Pull request merged**